packages/flame_tiled/lib/src/tile_atlas.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:ui';
+
+import 'package:flame/flame.dart';
+import 'package:flame/image_composition.dart';
+import 'package:flame/sprite.dart';
+import 'package:flame_tiled/src/rectangle_bin_packer.dart';
+import 'package:meta/meta.dart';
+import 'package:tiled/tiled.dart' as tiled;
+
+/// One image atlas for all Tiled image sets in a map.
+class TiledAtlas {
+  /// Single atlas for all renders.
+  // Retain this as SpriteBatch can dispose of the original image for flips.
+  final Image? atlas;
+
+  /// Map of all source images to their new offset.
+  final Map<String, Offset> offsets;
+
+  /// The single batch operation for this atlas.
+  final SpriteBatch? batch;
+
+  /// Image key for this atlas.
+  final String key;
+
+  /// Track one atlas for all images in the Tiled map.
+  TiledAtlas._(this.atlas, this.offsets, this.key)
```

Comment:
I'm assuming this is how you do private constructors in Dart? That's cool!

Maybe include a note to use the static "load" method in the comment?